lib/rules/jsx-boolean-value.js

```diff
@@ -17,44 +26,78 @@ module.exports = {
     },
     fixable: 'code',
 
-    schema: [{
-      enum: ['always', 'never']
-    }]
+    schema: {
+      anyOf: [{
+        type: 'array',
+        items: [{enum: [ALWAYS, NEVER]}],
+        additionalItems: false
+      }, {
+        type: 'array',
+        items: [{
+          enum: [ALWAYS]
+        }, {
+          type: 'object',
+          additionalProperties: false,
+          properties: {
+            [NEVER]: exceptionsSchema
+          }
+        }],
+        additionalItems: false
+      }, {
+        type: 'array',
+        items: [{
+          enum: [NEVER]
+        }, {
+          type: 'object',
+          additionalProperties: false,
+          properties: {
+            [ALWAYS]: exceptionsSchema
+          }
+        }],
+        additionalItems: false
+      }]
+    }
   },
 
   create: function(context) {
-    var configuration = context.options[0] || 'never';
+    const configuration = context.options[0] || NEVER;
+    const configObject = context.options[1] || {};
+    const exceptions = new Set((configuration === ALWAYS ? configObject[NEVER] : configObject[ALWAYS]) || []);
+    const exceptionProps = Array.from(exceptions, (name) => `\`${name}\``).join(', ');
+    const exceptionsMessage = exceptions.size > 0 ? ` for the following props: ${exceptionProps}` : '';
+    const data = {exceptionsMessage: exceptionsMessage};
 
-    var NEVER_MESSAGE = 'Value must be omitted for boolean attributes';
-    var ALWAYS_MESSAGE = 'Value must be set for boolean attributes';
+    const NEVER_MESSAGE = 'Value must be omitted for boolean attributes{{exceptionsMessage}}';
+    const ALWAYS_MESSAGE = 'Value must be set for boolean attributes{{exceptionsMessage}}';
 
     return {
-      JSXAttribute: function(node) {
-        switch (configuration) {
-          case 'always':
-            if (node.value === null) {
-              context.report({
-                node: node,
-                message: ALWAYS_MESSAGE,
-                fix: function(fixer) {
-                  return fixer.insertTextAfter(node, '={true}');
-                }
-              });
+      JSXAttribute(node) {
+        const propName = node.name && node.name.name;
+        const value = node.value;
+        const isException = exceptions.has(propName);
+
+        const isAlways = configuration === ALWAYS ? !isException : isException;
+        const isNever = configuration === NEVER ? !isException : isException;
```

Comment:
I find these two ternaries less clear than if/else. I had to work through the logic here, which slowed me down. It might be worth converting.